AK/IntrusiveList.h

```diff
@@ -382,7 +382,9 @@ inline T* IntrusiveList<T, Container, member>::node_to_value(SubstitutedIntrusiv
     //       named `member`.
     //       This function effectively takes in the address of the data member, and returns the address
     //       of the value (of type T) holding that member.
-    return bit_cast<T*>(bit_cast<unsigned char*>(&node) - bit_cast<unsigned char*>(member));
+    auto member_offset_sized = bit_cast<Conditional<sizeof(member) == 8, u64, u32>>(member);
+    auto member_offset = bit_cast<unsigned char*>(static_cast<uintptr_t>(member_offset_sized));
+    return bit_cast<T*>(bit_cast<unsigned char*>(&node) - member_offset);
```

Comment:
@nico Do you know if this is a valid implementation of ``container_of`` using pointer-to-member for the MSVC ABI? Ali mentioned that you'd given a talk about clang-cl that mentioned the odd constraints of pointer-to-member on Windows.

Reply:
https://docs.google.com/presentation/d/1oxNHaVjA9Gn_rTzX6HIpJHP7nXRua_0URXxxJ3oYRq0/edit?usp=drivesdk / https://youtu.be/rXi065XC6zY is the talk. (Might look at your pr later.)

Reply:
Thanks! Looks like the relevant info was on slides 41-45. I think that since we're not currently making intrusive lists of classes with multiple inheritance (I think?) and have essentially no virtual inhertiance, this should be fine? It feels like when it doesn't work, we'll know very loudly.